### PR TITLE
Retry restore job if the connection hasn't reflected the latest moco-admin's privileges yet

### PR DESCRIPTION
**Pull request by ysksuzuki**
#284

### CODE REVIEW

File: cmd/moco-backup/cmd/root.go [hunk 1]
```diff
@@ -63,6 +64,17 @@ var rootCmd = &cobra.Command{
 // Execute adds all child commands to the root command and sets flags appropriately.
 // This is called by main.main(). It only needs to happen once to the rootCmd.
 func Execute() {
+	defer func() {
+		err := recover()
+		if err != nil && errors.Is(err.(error), backup.ErrBadConnection) {
```
Comment (ymmt2005) [inline cmd/moco-backup/cmd/root.go:69]: if not, the error needs to be rethrown with panic.  Here's an example.
https://cs.opensource.google/go/go/+/refs/tags/go1.16.6:src/encoding/json/encode.go;l=322

**Review by ymmt2005 — CHANGES_REQUESTED** [review]

File: controllers/mysqlcluster_controller.go [hunk 2]
```diff
@@ -1092,6 +1092,11 @@ func (r *MySQLClusterReconciler) reconcileV1RestoreJob(ctx context.Context, req
 				Resources: []string{"pods"},
 				Verbs:     []string{"get"},
 			},
+			{
```
Comment (ymmt2005) [inline controllers/mysqlcluster_controller.go:1095]: How does this relate to the issue #284?

Comment (ysksuzuki) [reply controllers/mysqlcluster_controller.go:1095]: It's nothing to do with it. I just happened to find that the restore job couldn't create events, and fixed it. Should I raise another PR?

Comment (ymmt2005) [reply controllers/mysqlcluster_controller.go:1095]: Yes, please.

**Review by ymmt2005 — APPROVED** [review]
LGTM

**Pull request merged**